src/scripts/main.js

```diff
@@ -1,7 +1,52 @@
 'use strict';
 
-const pushNotification = (posTop, posRight, title, description, type) => {
-  // write code here
+const createNotification = (posTop, posRight, title, description, type) => {
+  const notification = document.createElement('div');
+
+  notification.className = `notification ${type}`;
+  notification.style.position = 'fixed';
+  notification.style.top = `${posTop}px`;
+  notification.style.right = `${posRight}px`;
+
+  const titleElement = document.createElement('h2');
+
+  titleElement.className = 'title';
+  titleElement.textContent = title;
+  notification.appendChild(titleElement);
+
+  const descriptionElement = document.createElement('p');
+
+  descriptionElement.textContent = description;
+  notification.appendChild(descriptionElement);
+
+  return notification;
+};
+
+const showNotification = (notification, duration) => {
+  document.body.appendChild(notification);
+
+  setTimeout(() => {
+    document.body.removeChild(notification);
+  }, duration);
+};
+
+const pushNotification = (
+  posTop,
+  posRight,
+  title,
+  description,
+  type,
+  duration = 2000,
+) => {
+  const notification = createNotification(
+    posTop,
+    posRight,
+    title,
+    description,
+    type,
+  );
+
+  showNotification(notification, duration);
 };
```

Comment:
This function 'pushNotification' could be considered as business logic, as it uses the UI functions to achieve the desired functionality. This function could remain in the 'main.js' file.